Input and DB consistency

njharman opened the pull request:
The global "con" was closed in some functions. It was shadowed by local
variable "con". There was lots of identical or very similar code.

  - Ignore case for inputs where it makes sense
  - Change datetime to date column types, use dates
  - Move if table exists logic into create_table function
  - Move "get unique values" from python "set" to database "DISTINCT"
  - Cut cur global
  - Use con global consistently
  - Deduplicate "show records" code
  - Deduplicate PlayRecord's prettytable and column code
  - Add input_ functions to make input consistent and remove code duplication
  - Uppercase date_format constant global
  - Lowercase Record variable
  - Replace empty print() with "\n"
  - Add count of existing records on start
  - Add main code body into a standard if __main__ cause

njharman committed:
Input and DB consistency

The global "con" was closed in some functions. It was shadowed by local
variable "con". There was lots of identical or very similar code.

  - Ignore case for inputs where it makes sense
  - Change datetime to date column types, use dates
  - Move if table exists logic into create_table function
  - Move "get unique values" from python "set" to database "DISTINCT"
  - Cut cur global
  - Use con global consistently
  - Deduplicate "show records" code
  - Deduplicate PlayRecord's prettytable and column code
  - Add input_ functions to make input consistent and remove code duplication
  - Uppercase date_format constant global
  - Lowercase Record variable
  - Replace empty print() with "\n"
  - Add count of existing records on start
  - Add main code body into a standard if __main__ cause

njharman committed:
Input and DB consistency

The global "con" was closed in some functions. It was shadowed by local
variable "con". There was lots of identical or very similar code.

  - Ignore case for inputs where it makes sense
  - Change datetime to date column types, use dates
  - Move if table exists logic into create_table function
  - Move "get unique values" from python "set" to database "DISTINCT"
  - Cut cur global
  - Use con global consistently
  - Deduplicate "show records" code
  - Deduplicate PlayRecord's prettytable and column code
  - Add input_ functions to make input consistent and remove code duplication
  - Uppercase date_format constant global
  - Lowercase Record variable
  - Replace empty print() with "\n"
  - Add count of existing records on start
  - Add main code body into a standard if __main__ cause

hongkongwargamer committed:
Merge branch 'main' into input

hongkongwargamer committed:
Ignored case for input where possible

hongkongwargamer commented:
Ignored case for input where possible.

hongkongwargamer committed:
Ignored case for input where possible

hongkongwargamer committed:
WIP : putting prettytable calls to a function

Most of it anyway, still working on it

hongkongwargamer committed:
Using SELECT DISTINCT to get unique values

hongkongwargamer committed:
Clean up con/ cur.  Put a count of records on

Working on having record properly deleted

hongkongwargamer committed:
Deletion working properly

hongkongwargamer commented:
- DeDuplicated PrettyTable and Column code
- Cleaned up the con & cur
- Using SELECT DISTINCT

hongkongwargamer commented:
- Upper case date format constant
- Lower case record variable
- Main body in If Main clause
- Program quits properly

njharman committed:
Merge branch 'input' of github.com:njharman/ASLGameLog into input

njharman committed:
Fix merge issues.

rename .gitignore  (has to have no extension to be recognized)

Files listed in .gitignore will be ignored. Changes/additions will not
be applied to repo.

Add database and csv to .gitignore
Cut License(sic), requirements.txt from .gitignore

Want project files such as LICENSE and requirements.txt to be part
of repo so people get them when they check out repo.
Don't want output and other files that will vary by user checked into
repo.

pip -Ur requirements.txt (will install all the requirements for this
project)
